sandbox/grist/acl.py

```diff
@@ -108,32 +144,32 @@ def prepare_acl_col_renames(docmodel, useractions, col_renames_dict):
         log.warning("Error examining aclRule: %s", e)
 
   # Go through again checking if anything in ACL formulas is affected by the rename.
-  for rule_rec in docmodel.aclRules.all:
-    if rule_rec.aclFormula:
-      formula = rule_rec.aclFormula
-      patches = []
-
-      for entity in parse_acl_grist_entities(rule_rec.aclFormula):
-        if entity.type == 'recCol':
-          table_id = docmodel.aclResources.table.get_record(int(rule_rec.resource)).tableId
-        elif entity.type == 'userAttrCol':
-          table_id = user_attr_tables.get(entity.extra)
-        else:
-          continue
-        col_id = entity.name
-        new_col_id = col_renames_dict.get((table_id, col_id))
-        if not new_col_id:
-          continue
-        patch = textbuilder.make_patch(
-            formula, entity.start_pos, entity.start_pos + len(entity.name), new_col_id)
-        patches.append(patch)
-
-      replacer = textbuilder.Replacer(textbuilder.Text(formula), patches)
-      txt = replacer.get_text()
-      rule_updates.append((rule_rec, {'aclFormula': txt,
-                                      'aclFormulaParsed': parse_predicate_formula_json(txt)}))
-
-  def do_renames():
-    useractions.doBulkUpdateFromPairs('_grist_ACLResources', resource_updates)
-    useractions.doBulkUpdateFromPairs('_grist_ACLRules', rule_updates)
-  return do_renames
+  for rule_rec in useractions.docmodel.aclRules.all:
+
+    if not rule_rec.aclFormula:
+      continue
+    formula = rule_rec.aclFormula
+
+    def renamer(subject):
+      if subject.type == 'recCol':
+        table_id = useractions.docmodel.aclResources.table.get_record(int(rule_rec.resource)).tableId
+      elif subject.type == 'userAttrCol':
+        table_id = user_attr_tables.get(subject.extra)
+      else:
+        return None
+      col_id = subject.name
+      return col_renames_dict.get((table_id, col_id))
+
+    new_acl_formula = predicate_formula.process_renames(formula, _ACLEntityCollector(), renamer)
+    new_rule_record = {"aclFormula": new_acl_formula}
+    try:
+      new_rule_record["aclFormulaParsed"] = parse_predicate_formula_json(new_acl_formula)
+    except SyntaxError:
+      pass
```

Comment:
I don't believe this can result in inconsistency. Even if `aclFormula` is updated, it will still be broken if it originally was. The parsed version will stay obsolete and broken until the text version is fixed by the user, when the parsed version is re-generated and fixed thereof.

But I went through the logic again and recalled that the new `process_renames` function doesn't do anything to a formula with syntax errors, so nothing will actually happen. Error checking here seems indeed redundant. I removed them and left comments to explain the situation.

Also: I have a test case for this behavior (don't process syntactically wrong formulas), so we should be safe.